src/finch/tensor.py

```diff
@@ -383,11 +383,12 @@ def to_storage(self, storage: Storage) -> "Tensor":
         return Tensor(self._from_other_tensor(self, storage=storage))
 
     @classmethod
-    def _from_other_tensor(cls, tensor: "Tensor", storage: Storage | None) -> JuliaObj:
+    def _from_other_tensor(cls, tensor: "Tensor", storage: Storage) -> JuliaObj:
         order = cls.preprocess_order(storage.order, tensor.ndim)
-        return jl.swizzle(
-            jl.Tensor(storage.levels_descr._obj, tensor._obj.body), *order
+        result = jl.copyto_b(
+            jl.swizzle(jl.Tensor(storage.levels_descr._obj), *order), tensor._obj
         )
+        return jl.dropfills(result) if tensor._is_dense else result
```

Comment:
Does this always copy? We should avoid a copy if at all possible.

Reply:
Are you asking if `dropfills` always copy?

Reply:
I'm asking if `copyto_b` always copies.

Reply:
Yes, I think `copyto_b` always copies.

Reply:
I think `copyto_b` is used here correctly. It's here to implement `to_storage` that is meant to copy Tensor to a new Storage.

Reply:
Is there a way to avoid the copy in some cases, e.g. when the format is the same?

Reply:
I can include it in `asarray` later, once we can compare tensors formats. Right now there's no way to know if a given tensor is `csr` or `csc` so we don't know if the `format` argument is the same.

Reply:
What I'm worried about is `xp.asarray` always copying.